src/common/blkdev.cc

```diff
@@ -494,7 +482,24 @@ int block_device_run_smartctl(const char *device, int timeout,
 #elif defined(__APPLE__)
 #include <sys/disk.h>
 
-int get_block_device_size(int fd, int64_t *psize)
+const char *BlkDev::sysfsdir() const {
+  assert(false);  // Should never be called on Apple
```

Comment:
probably `ceph_assert` instead?

Reply:
i think `ceph_assert()` is for something could happen. and `assert()` is for debugging. so i am inclined to keep as it is. also, i think `blkdev.cc` is relatively a low-level component. ideally it should have minimal dependencies. and would be better if it does not depend on logging and global ceph context which are indirectly pulled in as dependencies.

when looking at our building dependencies in `common`, i found that it's no way to figure out a dependency tree of compiling units: quite a lot of them are tightly coupled. we cannot fix this in one day/night. but we could try to avoid adding the arcs to this complicated dependency graph.